src/object/cli_obj.c

```diff
@@ -378,28 +379,121 @@ obj_grp_valid_shard_get(struct dc_object *obj, int idx,
 }
 
 static int
-obj_grp_shard_get(struct dc_object *obj, uint32_t grp_idx,
-		  uint64_t hash, unsigned int map_ver, uint32_t op)
+obj_select_leader_shard(struct dc_object *obj, uint32_t idx,
+			uint32_t *shard_idx)
+{
+	struct dc_obj_shard		*shard;
```

Comment:
This code looks eerily similar to ds_pool_select_leader_target.   Could we commonize this code?

Reply:
I will uniform the client-side and server-side leader election.